src/scope/FunctionHoister.js

```diff
@@ -51,7 +53,9 @@ class FunctionHoister {
   }
 
   registerParam(p) {
-    this.functionScope.register(p.name, new Variable(p));
+    const v = new Variable(p);
+    v.markDeclared();
+    this.functionScope.register(p.name, v);
```

Comment:
The code here is the same as in `hoistFunctionId()`.

We could combine these to a single function.